code/__HELPERS/test_reach.dm

```diff
@@ -0,0 +1,128 @@
+//test_reach(), by Deity Link
+//Basically sends a cheap projectile that moves from a turf A to a turf B in a "straight" line. Returns true if it reaches turf B without hitting anything on the way there (other than turf B itself)
+//Useful for instance to check if an item thrown/projectile fired from a A to B would reach its target
+
+/proc/test_reach(var/turf/origin,var/turf/destination,var/_pass_flags=0)
+	if (!origin || !destination)
+		return FALSE
+	if (origin.z != destination.z)
+		return FALSE
+	if (origin == destination)
+		return TRUE
+	var/obj/test_reach/test = new(origin,destination,_pass_flags)
+	var/result = test.main()
+	qdel(test)
+	return result
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/test_reach
+	invisibility = 101
+	pass_flags = 0
+
+	var/dist_x
+	var/dist_y
+	var/dx = 0
+	var/dy = 0
+	var/error = 0
+	var/target_angle = 0
+	var/bumped = 0
+	var/max_range = 100
+	var/turf/starting
+	var/turf/target
+
+	var/result = FALSE
+	var/finished = FALSE
+
+/obj/test_reach/New(var/turf/loc,var/turf/destination,var/_pass_flags)
+	..()
+	starting = loc
+	target = destination
+	pass_flags = _pass_flags
+
+/obj/test_reach/proc/main()
```

Comment:
```dm
var/vector/V = atoms2vector(loc, target)
V = V.normalized()
var/direction = vector2ClosestDir(V)
var/angle = V.toAngle()
var/turf/T = get_step(loc, direction)

if (T.density)
     switch(direction)
           if (NORTH)
                 // Dir2Angle_t is the trigonometric-wise angular count with EAST = 0, NORTH = 90, etc
                 var/north_east_angular_dist = abs(dir2angle_t(NORTHEAST) - angle)
                 var/north_west_angular_dist = abs(dir2angle_t(NORTHWEST) - angle)
                 // North-west closer to our angle
                 if (north_east_angular_dist > north_west_angular_dist)
                        // Move North-West

...
```

This is a bit ugly but this gives you an idea of how mathematising it avoid walking in random directions. I feel it's more elegant. I'm not sure which performs better, though.